drivers/pinmux/Makefile

```diff
@@ -6,20 +6,21 @@ ccflags-$(CONFIG_PINMUX_QMSI) += -I$(CONFIG_QMSI_INSTALL_PATH)/include
 obj-$(CONFIG_PINMUX_MCUX) += pinmux_mcux.o
 obj-$(CONFIG_PINMUX_STM32) += stm32/pinmux_stm32.o
 obj-$(CONFIG_PINMUX_BEETLE) += beetle/pinmux_board_v2m_beetle.o
-obj-$(CONFIG_BOARD_NUCLEO_F103RB) += stm32/pinmux_board_nucleo_f103rb.o
```

Comment:
The changes are just to put the nucleo boards in order.
As for pinmux, I will follow the include/dt-bindings convention, and I will leave the boards/ as it is in the rest of nucleos.